cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -87,14 +87,23 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 	if err != nil {
 		return nil, errors.Wrapf(err, "invalid SubnetAddressSpace %s", nc.SubnetAddressSpace)
 	}
-	subnet := cns.IPSubnet{
-		IPAddress:    primaryPrefix.Addr().String(),
-		PrefixLength: uint8(subnetPrefix.Bits()),
+
+	var subnet cns.IPSubnet
+	if nc.Type == v1alpha.VNETBlock {
+		subnet.IPAddress = nc.NodeIP
+		subnet.PrefixLength = uint8(subnetPrefix.Bits())
+	} else {
+		subnet.IPAddress = primaryPrefix.Addr().String()
+		subnet.PrefixLength = uint8(subnetPrefix.Bits())
 	}
 
 	req, err := createNCRequestFromStaticNCHelper(nc, primaryPrefix, subnet)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error while creating NC request from static NC")
 	}
+
+	if nc.Type == v1alpha.VNETBlock {
```

Comment:
can this block be moved to createNCRequestFromStaticNCHelper?

Reply:
The Request is getting created in the helper on the return statement for common code between Overlay and VnetScale. So after that seemed the right fit to update which would be this line, so that it happens for both Windows and Linux.

Reply:
could just create that req top of the func since secondaryIPConfigs is the only thing that gets updated, then update  with  nc.NodeIP in the VNETBlock

Reply:
Trying to remove this block and set the value in the createNCRequestFromStaticNCHelper method for both overlay and Vnet Scale